universal-application-tool-0.0.1/app/auth/UatProfileAdapter.java

```diff
@@ -42,11 +43,17 @@ public UatProfileAdapter(
 
   protected abstract String emailAttributeName();
 
+  protected abstract ImmutableSet<Roles> roles();
+
   /** Merge the two provided profiles into a new UatProfileData. */
   public UatProfileData mergeUatProfile(UatProfile uatProfile, OidcProfile oidcProfile) {
     String emailAddress = oidcProfile.getAttribute(emailAttributeName(), String.class);
     uatProfile.setEmailAddress(emailAddress).join();
     uatProfile.getProfileData().addAttribute(CommonProfileDefinition.EMAIL, emailAddress);
+    // Meaning: whatever you signed in with most recently is the role you have.
+    for (Roles role : roles()) {
+      uatProfile.getProfileData().addRole(role.toString());
```

Comment:
Question: this is just for my understanding, is this something in the cookies? or is it stored in the db? I didn't quite understand the root cause of the issue.

Reply:
Yep, all in the cookie - not persisted.  Clearing the cookie didn't fix it since it wasn't being set at all if the account already existed.